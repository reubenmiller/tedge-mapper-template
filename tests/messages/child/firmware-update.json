{
    "agentId": "612000655",
    "c8y_Firmware": {
        "name": "iot-linux",
        "url": "https://example.com",
        "version": "1.0.0"
    },
    "creationTime": "2023-05-25T08:58:19.504Z",
    "delivery": {
        "log": [],
        "status": "PENDING",
        "time": "2023-05-25T08:58:19.927Z"
    },
    "description": "Update firmware to: \"iot-linux\" (version: 1.0.0)",
    "deviceId": "862000795",
    "externalSource": {
        "externalId": "sim_tedge01_child01",
        "type": "c8y_Serial"
    },
    "id": "2021340",
    "status": "PENDING"
}