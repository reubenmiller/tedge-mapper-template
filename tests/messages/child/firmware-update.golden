Route: c8y-devicecontrol-notifications (c8y/devicecontrol/notifications)

Input Message
  topic:    c8y/devicecontrol/notifications

Output Message (mqtt)
  topic:    c8y/devicecontrol/notifications/sim_tedge01_child01/c8y_Firmware

    {
      "_ctx": {
        "agentID": "612000655",
        "deviceID": "862000795",
        "id": "2021340",
        "localSerial": "child01",
        "lvl": 1,
        "opType": "c8y_Firmware",
        "operationID": "2021340",
        "parent": "sim_tedge01",
        "serial": "sim_tedge01_child01"
      },
      "payload": {
        "agentId": "612000655",
        "c8y_Firmware": {
          "name": "iot-linux",
          "url": "https://example.com",
          "version": "1.0.0"
        },
        "creationTime": "2023-05-25T08:58:19.504Z",
        "delivery": {
          "log": [],
          "status": "PENDING",
          "time": "2023-05-25T08:58:19.927Z"
        },
        "description": "Update firmware to: \"iot-linux\" (version: 1.0.0)",
        "deviceId": "862000795",
        "externalSource": {
          "externalId": "sim_tedge01_child01",
          "type": "c8y_Serial"
        },
        "id": "2021340",
        "status": "PENDING"
      }
    }

Route: firmware-update-operation (c8y/devicecontrol/notifications/+/c8y_Firmware)

Input Message
  topic:    c8y/devicecontrol/notifications/sim_tedge01_child01/c8y_Firmware

Output Message (mqtt)
  topic:    tedge/child01/commands/firmware_update/start

    {
      "device": "child01",
      "id": "2021340",
      "name": "iot-linux",
      "sha256": null,
      "url": "https://example.com",
      "version": "1.0.0"
    }

Route: simulation - firmware plugin - send to a device (tedge/commands/firmware_update/start, tedge/+/commands/firmware_update/start)

Input Message
  topic:    tedge/child01/commands/firmware_update/start

Output Message (mqtt)
  topic:    tedge/child01/commands/req/firmware_update

    {
      "attempt": 1,
      "device": "child01",
      "id": "2021340",
      "name": "iot-linux",
      "sha256": null,
      "url": "https://example.com",
      "version": "1.0.0"
    }

Route: simulation - device - set to executing (tedge/commands/req/firmware_update, tedge/+/commands/req/firmware_update)

Input Message
  topic:    tedge/child01/commands/req/firmware_update

Output Message (mqtt)
  topic:    tedge/child01/commands/res/firmware_update

    {
      "attempt": 1,
      "device": "child01",
      "id": "2021340",
      "name": "iot-linux",
      "sha256": null,
      "status": "executing",
      "url": "https://example.com",
      "version": "1.0.0"
    }

Route: simulation - device - set to successful (tedge/commands/res/firmware_update, tedge/+/commands/res/firmware_update)

Input Message
  topic:    tedge/child01/commands/res/firmware_update

Output Message (mqtt)
  topic:    tedge/child01/commands/firmware_update/done/successful

    {
      "attempt": 1,
      "device": "child01",
      "id": "2021340",
      "name": "iot-linux",
      "sha256": null,
      "status": "successful",
      "url": "https://example.com",
      "version": "1.0.0"
    }

Route: firmware-update-operation-done (tedge/+/commands/firmware_update/done/failed, tedge/+/commands/firmware_update/done/successful)

Input Message
  topic:    tedge/child01/commands/firmware_update/done/successful

Output Updates
  topic:    c8y/s/us/sim_tedge01_child01

115,iot-linux,1.0.0,https://example.com


Output Message (mqtt)
  topic:    tedge/child01/commands/firmware_update/done

    {
      "attempt": 1,
      "device": "child01",
      "id": "2021340",
      "name": "iot-linux",
      "sha256": null,
      "status": "successful",
      "url": "https://example.com",
      "version": "1.0.0"
    }

Route: set-operation-to-done (tedge/+/commands/+/done)

Input Message
  topic:    tedge/child01/commands/firmware_update/done

Output Message (api)
  request:  PUT /devicecontrol/operations/2021340

    {
      "attempt": 1,
      "device": "child01",
      "name": "iot-linux",
      "sha256": null,
      "status": "SUCCESSFUL",
      "url": "https://example.com",
      "version": "1.0.0"
    }